File: iOS9-ReviewMemory-Objc/iOS9-ReviewMemory-Objc/ViewController.m
//
//  ViewController.m
//  iOS9-ReviewMemory-Objc
//
//  Created by Paul Solt on 11/14/19.
//  Copyright © 2019 Lambda, Inc. All rights reserved.
//

#import "ViewController.h"
#import "LSIPhoto.h"

@interface ViewController ()

@property NSMutableArray *apodArray;

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	_apodArray = [[NSMutableArray alloc] init];
	
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"November-APOD" ofType:@"json"];
	NSData *jsonData = [NSData dataWithContentsOfFile:filePath];
	NSError *error = nil;
	NSArray *photoArray = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
	
	if (error) {
		NSLog(@"Error: %@", error);
	}
	
	NSLog(@"APOD: %@", photoArray);
	
	// Loop through each dictionary
		// Create an object
		// Store in an array
	
	for (NSDictionary *photoDictionary in photoArray) {
		
		LSIPhoto *photo = [[LSIPhoto alloc] initWithDictionary:photoDictionary];
		
		// store the data
		[self.apodArray addObject:photo];
	}
	
	NSLog(@"Photos: %@", self.apodArray);
}


@end

File: iOS9-ReviewMemory/iOS9-ReviewMemory/ViewController.swift
//
//  ViewController.swift
//  iOS9-ReviewMemory
//
//  Created by Paul Solt on 11/14/19.
//  Copyright © 2019 Lambda, Inc. All rights reserved.
//

import UIKit


class ViewController: UIViewController {

	override func viewDidLoad() {
		super.viewDidLoad()
		
		
		
		NSURL *baseURL = [NSURL URLWithString:@"https://www."];
		NSURLComponents *components = [NSURLComponents componentsWithURL:baseURL resolvingAgainstBaseURL:true];
		NSURLQueryItem *artistItem = [NSURLQueryItem queryItemWithName:@"s" value:name];
		components.queryItems = @[artistItem];
		
		NSURL *requestURL = components.URL;
		NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:requestURL];
		
		[[[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
			
			if (error) {
				NSLog(@"Error fetching artist: %@.", error);
				completion(nil);
				return;
			}
			
			if (!data) {
				NSLog(@"No data returned from data task");
				completion(nil);
				return;
			}
			
			error = nil;
			NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
			if (error) {
				NSLog(@"Error decoding JSON from data: %@", error);
				completion(nil);
				return;
			}
			
			if (!dictionary || ![dictionary isKindOfClass:[NSDictionary class]]) {
				NSLog(@"Error: Expected top level dictionary in JSON result: %@", error);
				completion(nil);
				return;
			}

	}


}

